src/runtime/lpython/lpython.py

```diff
@@ -498,3 +498,120 @@ def ccallable(f):
+
+        # Get the source code of the function
+        source_code = getsource(function)
+        source_code = source_code[source_code.find('\n'):]
+
+        # Create a filename based on the function name
+        # filename = function.__name__ + ".py"
+
+        # Open the file for writing
+        with open("a.py", "w") as file:
+            # Write the Python source code to the file
+            file.write("@ccallable")
+            file.write(source_code)
+
+        self.fn_name = function.__name__
+        template = f"""// Python headers
+#include <Python.h>
+
+// NumPy C/API headers
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION // remove warnings
+#include <numpy/ndarrayobject.h>
+
+// Generated C code
+#include "a.h"
+
+// Define the Python module and method mappings
+static PyObject* module_add_numbers(PyObject* self, PyObject* args) {{
+    // Initialize NumPy
+    import_array();
+
+    int size;
+    double result;
+    double *array;
+    PyObject *py_array;
+    struct r64 *s_array = malloc(sizeof(struct r64));
+
+    // Parse the arguments from Python
+    if (!PyArg_ParseTuple(args, "iO", &size, &py_array)) {{
+        return NULL;
+    }}
+
+    // Create C arrays from numpy objects:
+    PyArray_Descr *descr = PyArray_DescrFromType(NPY_DOUBLE);
+    npy_intp dims[1];
+    if (PyArray_AsCArray(&py_array, (void *)&array, dims, 1, descr) < 0) {{
+        PyErr_SetString(PyExc_TypeError, "error converting to c array");
+        return NULL;
+    }}
+
+    s_array->data = array;
+    s_array->n_dims = 1;
+    s_array->dims[0].lower_bound = 0;
+    s_array->dims[0].length = size;
+    s_array->is_allocated = false;
+
+    // Call the C function
+    result = {self.fn_name}(size, s_array);
+
+    // Build and return the result as a Python object
+    return Py_BuildValue("d", result);
+}}
+
+// Define the module's method table
+static PyMethodDef module_methods[] = {{
+    {{"{self.fn_name}", module_add_numbers, METH_VARARGS, "Add two numbers."}},
+    {{NULL, NULL, 0, NULL}}
+}};
+
+// Define the module initialization function
+static struct PyModuleDef module_def = {{
+    PyModuleDef_HEAD_INIT,
+    "my_module",
+    "A sample Python module that exposes a C function.",
+    -1,
+    module_methods
+}};
+
+PyMODINIT_FUNC PyInit_my_module(void) {{
+    PyObject* module;
+
+    // Create the module object
+    module = PyModule_Create(&module_def);
+    if (!module) {{
+        return NULL;
+    }}
+
+    return module;
+}}
+"""
+        with open("a.c", "w") as file:
+            # Write the C source code to the file
+            file.write(template)
+
+        r = os.system("lpython --show-c --disable-main a.py > a.h")
+        assert r == 0, "Failed to create C file"
+        gcc_flags = " -bundle -flat_namespace -undefined suppress "
+        python_path = "-I" "$CONDA_PREFIX/include/python3.10 "
+        numpy_path = "-I" "$CONDA_PREFIX/lib/python3.10/site-packages/" \
+            "numpy/core/include/"
```

Comment:
Is this okay? or is there any other way to specify the path for Numpy?

Reply:
For NumPy use `import numpy; print numpy.get_include()`.

Reply:
For the Python path, I think this works: `from distutils.sysconfig import get_python_inc; print get_python_inc()`.